test/test_spec.py

```diff
@@ -1117,6 +1118,155 @@ class _RootStrictSpec:
         )
         assert is_something_error(ValueError, validate_data_spec, nok_data_leaf, _RootStrictSpec)
 
+    def test_conditional_existence2(self):
+        class _CondExist:
+            a = Checker([COND_EXIST], optional=True, COND_EXIST=dict(ANY=['b']))
+            b = Checker([COND_EXIST], optional=True, COND_EXIST=dict(ANY=['a']))
+
+        assert validate_data_spec(dict(a=1, b=1), _CondExist)
+        assert validate_data_spec(dict(a=1), _CondExist)
+        assert validate_data_spec(dict(b=1), _CondExist)
```

Comment:
I think from the interface, the expected behavior is failing the `dict(a=1)` (not have ANY[b] in field `a` checker) and `dict(b=1)` (not have ANY[a] in field `b` checker).

But since the checks have condition `if any_keys and isinstance(value, UnknownFieldValue)` added, so if there's no `dict(d=1)` passed in the data the ANY checker will be by-passed.